Frontend/ui-customer/src/Flow.purs

```diff
@@ -105,89 +106,39 @@ import Screens.TicketBookingScreen.Transformer
+  when callInitUI $ lift $ lift $ initUI -- TODO:: Can we move this to Main
   when showSplashScreen $ toggleSplash true
-  _ <- lift $ lift $ liftFlow $ logEventWithParams logField_ "ny_user_app_version" "version" versionName
-  _ <- lift $ lift $ liftFlow $ logEvent logField_ "ny_user_entered_app"
-  if (getValueToLocalStore COUNTRY_CODE == "__failed") || (getValueToLocalStore COUNTRY_CODE == "(null)")
-  then do
-    setValueToLocalStore COUNTRY_CODE "+91"
-  else pure unit
-  if (getMerchant FunctionCall) == PASSCULTURE then setValueToLocalStore LANGUAGE_KEY $ getValueFromConfig "defaultLanguage"
-    else pure unit
-  verifyPerson gPayload
-
-
-verifyPerson :: GlobalPayload -> FlowBT String Unit
-verifyPerson globalPayload = do
-  case globalPayload ^. _payload ^. _signatureAuthData of
-    Just (SignatureAuthData signatureAuth) -> do
-      response <- lift $ lift $ Remote.triggerSignatureBasedOTP (SignatureAuthData signatureAuth)
-      case response of
-        Right (TriggerSignatureOTPResp triggerSignatureOtpResp) -> do
-              case triggerSignatureOtpResp.person of
-                Just (User person) -> do
-                  lift $ lift $ setLogField "customer_id" $ encode (person.id)
-                  setValueToLocalStore CUSTOMER_ID person.id
-                  mobileNumber <- liftFlowBT $ runEffectFn2 getMobileNumber signatureAuth.authData (fromMaybe "" person.maskedMobileNumber)
-                  setValueToLocalStore MOBILE_NUMBER mobileNumber
-                _ -> pure unit
-              case triggerSignatureOtpResp.token of
-                Just token -> setValueToLocalStore REGISTERATION_TOKEN token
-                Nothing -> pure unit
-              handleDeepLinks (Just globalPayload) false
-        Left err -> do
-          liftFlowBT $ pure $ runFn3 emitJOSEvent "java" "onEvent" $ encode $  EventPayload { event : "signature_auth_failed", payload : Nothing}
-          pure unit
-    Nothing -> if getValueToLocalStore REGISTERATION_TOKEN /= "__failed" && getValueToLocalStore REGISTERATION_TOKEN /= "(null)"
-                then handleDeepLinks (Just globalPayload) false
-                else if (showCarouselScreen FunctionCall) then welcomeScreenFlow else enterMobileNumberScreenFlow
+  tokenValidity <- validateToken signatureAuthData
+  if tokenValidity 
+    then currentFlowStatus 
+    else validateAuthData $ signatureAuthData
+  
+  where
+    signatureAuthData = 
+      gPayload ^._payload ^._signatureAuthData
+    validateAuthData signatureAuthData = 
+      case signatureAuthData of
+        Just signatureAuth -> do
+          response <- lift $ lift $ Remote.triggerSignatureBasedOTP signatureAuth
+          validationStatus <- validateSignaturePayload signatureAuth response
+          when validationStatus $ handleDeepLinks (Just gPayload) false
+        Nothing -> 
+          if showCarouselScreen FunctionCall -- TODO:: @vicky I think don't have to check for deeplinks here
```

Comment:
previously we will check signature auth first then call registration token. but now logic is changed so we have to add in this place.

 ```
if tokenValidity 
    then handleDeepLinks (Just gPayload) false 
    else validateAuthData $ signatureAuthData
```